txpool/account.go

```diff
@@ -117,15 +117,19 @@ func (m *accountsMap) allTxs(includeEnqueued bool) (
 		defer account.promoted.unlock()
 
 		if account.promoted.length() != 0 {
-			allPromoted[addr] = account.promoted.queue
+			// slice is not thread safe, so copy it would be reasonable
```

Comment:
copy may cause mem crease huge.

Reply:
My mistake. Slice assignment in go is a reference copy, which implements `copy on write`.

Test example:

```go
func main() {
	s := []int{0, 1, 2, 3}
	ModifySliceFunc(s)
	fmt.Printf("In main: s is %v, pointer to s is %p\n", s, &s)
	fmt.Printf("In main: s[3] = %v\n", s[3])
}
func ModifySliceFunc(s []int) {
	s = s[2:]
	s = append(s, 4)
	fmt.Printf("In ModifySliceFunc: s is %v, pointer to s is %p\n", s, &s)
}
// Output:
//In ModifySliceFunc: s is [2 3 4], pointer to s is 0xc0000ac030
//In main: s is [0 1 2 3], pointer to s is 0xc0000ac018
//In main: s[3] = 3
```